buildSrc/subprojects/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/SubprojectsInfo.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.buildutils.tasks
+
+import com.google.gson.GsonBuilder
+import gradlebuild.buildutils.model.GradleSubproject
+import org.gradle.api.DefaultTask
+import java.io.File
+
+open class SubprojectsInfo : DefaultTask() {
+
+    private
+    val subprojectsFolder = project.layout.projectDirectory.dir("subprojects")
+
+    protected
+    val subprojectsJson = project.layout.projectDirectory.file(".teamcity/subprojects.json")
+
+
+    protected
+    fun generateSubprojectsJson(): String {
+        val subprojects = generateSubprojects()
+        val gson = GsonBuilder().setPrettyPrinting().create()
+        return gson.toJson(subprojects) + '\n'
+    }
+
+    private
+    fun generateSubprojects(): List<GradleSubproject> {
+        return subprojectsFolder.asFile.listFiles(File::isDirectory)!!
+            .filter {
+                File(it, "build.gradle.kts").exists() ||
+                    File(it, "build.gradle").exists()
+            }
```

Comment:
Added this filter so that when you switch branches, and a project is added/removed, it checks for the existence of a git tracked file.